task6_Sychev_Alexander/src/search_handling.cpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include "search_handling.h"
+
+bool Search_handling::isWithinRadius(const City& city, const City& selectedCity, double radius, int norm)
+{
+	return calc_distance.calculateDistance(selectedCity.x, selectedCity.y, city.x, city.y, norm) <= radius;
+}
+
+
+std::vector<City> Search_handling::searchCities(const std::vector<City>& cities, const City& selectedCity, double radius, int norm)
+{
+    std::vector<City> resultCities;
+
+    // Find cities within the given radius using copy_if and back_inserter
```

Comment:
you are missing 1 step in finding the cities in radius, you should find vertical and horizontal rectangles, and then to find the intersection square, and to find a circle inside the square.